### PR TITLE
RIA-4938

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-4938

### Change description ###
RIA-4938 Create a 'LastModifiedApplication' backend Event field 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MakeAnApplicationHandler.java [hunk 1]
```diff
@@ -80,6 +89,17 @@ public boolean canHandle(PreSubmitCallbackStage callbackStage,
...
 
+        String applicant = userDetailsHelper.getLoggedInUserRoleLabel(userDetails).toString();
+        UserRole applicantRole = userDetailsHelper.getLoggedInUserRole(userDetails);
+
+        waFieldsPublisher.addLastModifiedApplication(asylumCase,
+                applicant,
+                applicationType,
+                applicationReason,
+                applicationEvidence.orElse(Collections.emptyList()),
+                "Pending",
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MakeAnApplicationHandler.java:100]: is this 'P' capital?

Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MakeAnApplicationHandler.java:100]: On creation, MakeAnApplication is defaulted with this decision - "Pending"

Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MakeAnApplicationHandler.java:100]: Probably it's a good idea to move this hardcoded word to MakeAnApplicationDecision enum

**Review by iguannago — APPROVED** [review]

**Review by bradsorour — DISMISSED** [review]
LGTM - this ticket will require intermediate definitions merged before deployment.

**Review by iguannago — DISMISSED** [review]

**Review by iguannago — APPROVED** [review]

**Review by bradsorour — DISMISSED** [review]
LGTM

**Review by bradsorour — APPROVED** [review]
LGTM

**Pull request merged**